platform/android/java/lib/src/org/godotengine/godot/input/GodotInputHandler.java

```diff
@@ -266,67 +286,99 @@ public void initInputDevices() {
 		}
 	}
 
+	private int assignJoystickIdNumber(int deviceId) {
+		int godotJoyId = 0;
+		while (mJoystickIds.containsValue(godotJoyId)) {
+			godotJoyId++;
+		}
+		mJoystickIds.put(deviceId, godotJoyId);
+		return godotJoyId;
```

Comment:
What about keeping track of the last assigned `godotJoyId` and just incrementing it each time.
Seems like even if we were to remove joystick ids mapping, there's no significant advantage to reusing a `godotJoyId`, might as well just keep incrementing them.

Reply:
If a device gets temporarily disconnected and then is either reconnected or replaced by a different device in the middle of a game it would be best if it were to show back up with the previously disconnected id. I'm thinking about couch multiplayer scenario.

Reply:
Actually, that could be sources for subtle bugs, don't you think? If a specific device type (e.g: gamepad) is disconnected and replaced by another device type (e.g: vr controllers), would the Godot engine (or the developer's game logic) be smart enough to realize a new device type is now using an old device id?

@BastiaanOlij Any thoughts?

Reply:
Well, the game engine receives a disconnect and the game, if the designer is taking this into account, will also receive this event. It is really a case of which would be more expected behavior from a game player? To me, if I was having a couch multiplayer session and a pad quit and we either replaced batteries or turned on a different pad, I would expect it to allow the affected player to continue playing.

Reply:
There is already a function for this if I understand the problem here right:
https://github.com/godotengine/godot/blob/master/core/input/input.h#L333

The joy ids refer directly to a list of joysticks with slots for each connected joystick. 

Honestly can't say I fully agree with the solution which is why I did something different for VR controllers (closer to what @m4gr3d suggested, just forever increment an id) but it is what it is.

Reply:
I'm not familiar with that code. Does it renumber the renumbering done in the Java shim?